USSF-ORBIT/ussf-portal-cms
# feat: Redirect to portal login page if not logged in

# SC-288

SC-288

## Proposed changes

This PR adds middleware to redirect unauthenticated users to the USSF portal log in page, with a redirect URL parameter which can be used to redirect back to the CMS upon login (this functionality will be completed in https://github.com/USSF-ORBIT/ussf-portal-client/pull/579). This is dependent on a new env var: `PORTAL_URL`

Other changes include:
- Adds a public `/api/sysinfo` endpoint to the CMS app
- Adds the PR template to this repo

---

## Reviewer notes

### Test different roles locally:
- Try to access the CMS without being logged in and verify you are redirected to the portal log in page instead of seeing the "You don't have access to this page." message.
- Log into the portal as a user without CMS access (`user1 / user1pass`) and try to navigate to the CMS. You should be redirected to the no access page.
- Log into the portal as a user with CMS access (`cmsuser / cmsuserpass`) and try to navigate to the CMS. You should be able to access it.
- Log out of the CMS and verify you are redirected to the portal log in page (and have been logged out of the portal as well).

## Setup

Testing this PR requires running both the CMS and portal app locally:

1. Make sure you have the portal repo checked out, and the environment variable set to its path (in the CMS repo):
```
export PORTAL_PATH=../ussf-portal-client/
```

2. Start the portal app and other services in docker:
```
yarn portal:up
```

3. Start the CMS app:
```
yarn build && yarn start
```

4. You should be able to access the CMS at http://localhost:3001/ and the portal at http://localhost:3000/

---

## Code review steps

### As the original developer, I have

- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):

### As code reviewer(s), I have

- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
